src/main/java/org/jenkinsci/plugins/durabletask/ProcessLiveness.java

```diff
@@ -94,14 +103,49 @@ private static boolean _isAlive(VirtualChannel channel, int pid, Launcher launch
             this.pid = pid;
         }
         @Override public Boolean call() throws RuntimeException {
-            // JNR-POSIX does not seem to work on FreeBSD at least, so using JNA instead.
+            // Try jnr's waitpid
+            // We need to painstakingly extract the function via reflection to
+            // avoid class-loading errors in the event that the available jnr
+            // library doesn't support waitpid.
+            //
+            // This silliness is equivalent to,
+            //     int[] code = new int[1];
+            //     return jnr.waitpid(pid, code, WaitFlags.WNOHANG.intValue()) != -1;
+            try {
+                ClassLoader loader = getClass().getClassLoader();
+                Object jnr = PosixAPI.jnr();
+
+                // Find waitpid
+                Method waitpidMethod = jnr.getClass().getDeclaredMethod("waitpid", int.class, int[].class, int.class);
+
+                // Find WaitFlags.WNOHANG.intValue()
+                Class WaitFlags = loader.loadClass("jnr.constants.platform.WaitFlags");
+                Method intValue = WaitFlags.getDeclaredMethod("intValue");
+                Integer WNOHANG = (Integer) intValue.invoke(Enum.valueOf(WaitFlags, "WNOHANG"));
+
+                int[] code = new int[1];
+                waitpidMethod.setAccessible(true); // the implementations are package-local
+                Integer ret = (Integer) waitpidMethod.invoke(jnr, pid, code, WNOHANG);
+                return ret != -1;
+            } catch (InvocationTargetException x) {
+                LOGGER.log(Level.FINE, "jnr doesn't have waitpid support", x);
+            } catch (IllegalAccessException x) {
+                LOGGER.log(Level.FINE, "jnr doesn't have waitpid support", x);
+            } catch (ClassNotFoundException x) {
+                LOGGER.log(Level.FINE, "jnr doesn't have waitpid support", x);
+            } catch (NoSuchMethodException x) {
+                LOGGER.log(Level.FINE, "jnr doesn't have waitpid support", x);
+            }
```

Comment:
maybe also SecurityException, but it's not something going to work well in Jenkins anyway

Reply:
I've not encountered this one but we're now catching it anyways.